flasgger/base.py

```diff
@@ -91,14 +98,124 @@ class APISpecsView(MethodView):
     The /apispec_1.json and other specs
     """
     def __init__(self, *args, **kwargs):
-        view_args = kwargs.pop('view_args', {})
-        self.config = view_args.get('config')
-        self.spec = view_args.get('spec')
-        self.process_doc = view_args.get('sanitizer', BR_SANITIZER)
-        self.template = view_args.get('template')
-        self.definition_models = view_args.get('definition_models')
+        self.loader = kwargs.pop('loader')
         super(APISpecsView, self).__init__(*args, **kwargs)
 
+    def get(self):
+        """
+        The Swagger view get method outputs to /apispecs_1.json
+        """
+        return jsonify(self.loader())
+
+
+class SwaggerDefinition(object):
+    """
+    Class based definition
+    """
+    def __init__(self, name, obj, tags=None):
+        self.name = name
+        self.obj = obj
+        self.tags = tags or []
+
+
+class Swagger(object):
+
+    DEFAULT_CONFIG = {
+        "headers": [
+        ],
+        "specs": [
+            {
+                "endpoint": 'apispec_1',
+                "route": '/apispec_1.json',
+                "rule_filter": lambda rule: True,  # all in
+                "model_filter": lambda tag: True,  # all in
+            }
+        ],
+        "static_url_path": "/flasgger_static",
+        # "static_folder": "static",  # must be set by user
+        "swagger_ui": True,
+        "specs_route": "/apidocs/",
+        "doc_dir": "./docs/api/"
```

Comment:
Doc-parsing is new and presumably optional and opt-in, but `doc_dir` is being set in the `DEFAULT_CONFIG` here.  As a result all existing examples are passing through the new code in `get_specs`, and the `doc_dir=None` is not having any effect.

So I guess I'm checking whether it is really intended to set this as a default here.  If not, and especially if all other options are mentioned here, it might be OK to leave it commented per `static_folder`.

If the intention is to make this new feature default-on with some default path to the docs, then the `parse` option would need review.

Reply:
Doc auto loading is a optional feature, although it won't affect flasgger's other feature if there is no `./docs/api` in local directory, for more clearly, I'll remove the `doc_dir` config in `DEFAULT_CONFIG`